### PR TITLE
[WIP] Rename Comment to ReviewComment

### LINKED ISSUE #1511: Rename PulRequest.Comment client to PullRequest.ReviewComment
To better align with the [API docs](https://developer.github.com/v3/pulls/comments/), the access to the `PullRequestReviewClient` should be renamed from `client.PullRequest.Comment` to `client.PullRequest.ReviewComment` (also applies to the `Observable` versions of the client)

Normal octokit.net deprecation procedures should be followed (old variables being marked `[Obsolete]` for at least 2 releases before being deleted).

### PR DESCRIPTION
**Pull request by bmeverett**
Fixes  #1511  

Change references of Comment to ReviewComment based on the API change 

- [x] Rename Comment to ReviewComment in the PullRequestClient and interface as well as all tests related to it. 
- [x] Replace the Observable interfaces and classes

### CODE REVIEW

**Comment by bmeverett** [comment]
I changed this in the `PullRequestClient` as well as the `ObservablePullRequestClient` Should we change it anywhere else at this time?

File: Octokit.Reactive/Clients/ObservablePullRequestsClient.cs [hunk 1]
```diff
@@ -18,15 +18,17 @@ public class ObservablePullRequestsClient : IObservablePullRequestsClient
         /// <summary>
         /// Client for managing comments.
         /// </summary>
-        public IObservablePullRequestReviewCommentsClient Comment { get; private set; }
+        [Obsolete("Please use ObservablePullRequestsClient.ReviewComment. This will be removed in a future version")]
+        public IObservablePullRequestReviewCommentsClient Comment { get;  set; }
```
Comment (ryangribble) [inline Octokit.Reactive/Clients/ObservablePullRequestsClient.cs]: We still need the `Comment` endpoint to function so let's make it a getter only, and have it return the new client

eg
``` csharp
public IObservablePullRequestReviewCommentsClient Comment { get { return this.ReviewComment; } }
```

File: Octokit.Reactive/Clients/ObservablePullRequestsClient.cs [hunk 2]
```diff
@@ -18,15 +18,17 @@ public class ObservablePullRequestsClient : IObservablePullRequestsClient
...
+        public IObservablePullRequestReviewCommentsClient ReviewComment { get; private set; }
```
Comment (ryangribble) [inline Octokit.Reactive/Clients/ObservablePullRequestsClient.cs]: Same comments about whitespace and XmlDoc here

File: Octokit/Clients/IPullRequestsClient.cs [hunk 3]
```diff
@@ -15,7 +16,12 @@ public interface IPullRequestsClient
         /// <summary>
         /// Client for managing comments.
         /// </summary>
+        [Obsolete("Please use IPullRequestsClient.ReviewComment instead. This method will be removed in a future version")]
         IPullRequestReviewCommentsClient Comment { get; }
+        /// <summary>
```
Comment (ryangribble) [inline Octokit/Clients/IPullRequestsClient.cs]: fix formatting (whitespace) here for consistency

File: Octokit/Clients/IPullRequestsClient.cs [hunk 4]
```diff
@@ -15,7 +16,12 @@ public interface IPullRequestsClient
...
+        /// Client for managing comments.
```
Comment (ryangribble) [inline Octokit/Clients/IPullRequestsClient.cs]: I guess this comment should say "Client for managing review comments" now?

File: Octokit/Clients/PullRequestsClient.cs [hunk 5]
```diff
@@ -14,14 +15,18 @@ public class PullRequestsClient : ApiClient, IPullRequestsClient
...
+            ReviewComment = new PullRequestReviewCommentsClient(apiConnection);
         }
 
         /// <summary>
         /// Client for managing comments.
         /// </summary>
-        public IPullRequestReviewCommentsClient Comment { get; private set; }
-
+        [Obsolete("Please use PullRequestsClient.ReviewComment instead. This method will be removed in a future version")]
+        public IPullRequestReviewCommentsClient Comment { get;  set; }
```
Comment (ryangribble) [inline Octokit/Clients/PullRequestsClient.cs]: same comments around getter only, whitespace, and xmldoc wording change

**Comment by bmeverett** [comment]
@ryangribble I have made the requested changes.

**Comment by ryangribble** [comment]
Great work...  Many thanks for your contribution, this is good to go!

**Comment by ryangribble** [comment]
release_notes: Rename PullRequest.Comment => PullRequest.ReviewComment for clarity

**Pull request merged**